### ISSUE TITLE
Flux (influxDB2) : Grafana ignoring query results if "_value" is present

### ISSUE THREAD

**Issue by cgfoed** [grafana/grafana#30085]
I can't imagine this bug wasn't filed before but I looked through the issues and found nothing like it (?)

If your query produces a table that looks like that:
![image](https://user-images.githubusercontent.com/53257619/103791617-f1386280-5042-11eb-8a5a-5ae98a055f90.png)

The resulting data array will only contain the "_value" column.
![image](https://user-images.githubusercontent.com/53257619/103791910-4eccaf00-5043-11eb-853f-1e905627d77d.png)

Adding a simple rename command at query end resolves the issue:
|> rename(columns: {_value: "_value_omd"})

![image](https://user-images.githubusercontent.com/53257619/103792075-82a7d480-5043-11eb-8d1c-5dd133736824.png)

will result in a complete data array:
![image](https://user-images.githubusercontent.com/53257619/103792357-d5818c00-5043-11eb-8ed1-e70abc73199a.png)

**Comment by ryantxu** [comment]
Can you post a snippit of your results so we can add tests for the conversion?  See:
https://github.com/grafana/grafana/blob/master/pkg/tsdb/influxdb/flux/testdata/buckets.csv

This lets us write a test to make sure it gets to the correct format:
https://github.com/grafana/grafana/blob/master/pkg/tsdb/influxdb/flux/testdata/boolean.golden.txt

**Comment by cgfoed** [comment]
I realized this only can happen if you aggregate data within the query, e.g. by joining _(I guess pivots will work as well?)_
As requested I generated testdata spanning from 2021-01-07T07:21:50Z to 2021-01-07T07:38:10Z and attached it.

First CSV is by querying with the basics:

```
from(bucket: "temptest")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "GrafanatestData")
  |> filter(fn: (r) => r["_field"] == "A" or r["_field"] == "B" or r["_field"] == "C" or r["_field"] == "D" or r["_field"] == "E")
  |> aggregateWindow(every: v.windowPeriod, fn: mean, createEmpty: false)
  |> yield(name: "mean")
```
Result: [Query1_Raw_data.csv.txt](https://github.com/grafana/grafana/files/5780235/Query1_Raw_data.csv.txt)


Second CSV is with the following query:
```
A_DATA = from(bucket: "temptest")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "GrafanatestData")
  |> filter(fn: (r) => r["_field"] == "A")
  |> aggregateWindow(every: v.windowPeriod, fn: mean, createEmpty: false)

B_DATA = from(bucket: "temptest")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "GrafanatestData")
  |> filter(fn: (r) => r["_field"] == "B")
  |> aggregateWindow(every: v.windowPeriod, fn: mean, createEmpty: false)

C_DATA = from(bucket: "temptest")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "GrafanatestData")
  |> filter(fn: (r) => r["_field"] == "C")
  |> aggregateWindow(every: v.windowPeriod, fn: mean, createEmpty: false)

AB_DATA = join(
  tables: {A_temp:A_DATA, b_temp:B_DATA},
  on: ["_time", "_stop", "_start"]
)

join(
  tables: {AB_final:AB_DATA, C_final:C_DATA},
  on: ["_time", "_stop", "_start"]
)
|> drop(columns: ["_stop", "_start"])
```
Result: [Query2.csv.txt](https://github.com/grafana/grafana/files/5780236/Query2.csv.txt)


As you can see, there is a "_value" named column, which will break the importing of the data.

If you add a rename at the end of the query, Grafana will display **all** data, instead of just this column.
`|> rename(columns: {_value: "_value_C"})`

I guess this is an edge-case, but it was a major head scratcher for me as I need to work with joined tables

**Comment by cgfoed** [comment]
@ryantxu does this still 'needs more info' ?

**Comment by gabor** [comment]
just summarising the problem: when the flux-response arrives, grafana checks if there is a column named "_value":
  - if yes, grafana only uses data from that column
  - if no, grafana uses data from every column

the problem happens when the flux-response has multiple columns with data, and one is called "_value". for example, if the columns are named "_value", "_value_2"; grafana will use "_value" and ignore "_value2".

**Comment by gabor** [comment]
i understand this can be unintuitive, unfortunately however we change the handling of the data, we need to do it in a backward-compatible way, and i don't see an easy way to achieve that. we'll mark this issue as a nice-to-have feature.

**Comment by philomory** [comment]
> unfortunately however we change the handling of the data, we need to do it in a backward-compatible way
@gabor I don't think this is accurate. Flux support in Grafana is explicitly in beta right now, so this is _absolutely_ the time to fix issues even if it breaks backwards compatibility. That's basically what beta _means_.

**Comment by cgfoed** [comment]
not sure to what exactly this flux-specific code needs to be backwards compatible? flux is in its first iteration at the moment imho

**Comment by gabor** [comment]
@cgfoed what i mean is, if someone already created a dashboard where (probably by mistake) they have in the data both `_value` and `_value2`, and they do not know about this problem, the dashboard might currently work for them. but, if we change this to be more predictable, the data-returned will be different, possibly breaking the dashboard.

but, @philomory has a good point, Flux-support is still in beta, so we will probably be willing to break backward-compatibility to make this more consistent.

**Comment by gabor** [comment]
when https://github.com/grafana/grafana/issues/32842 happens it should fix this issue.

**Comment by gabor** [comment]
@cgfoed @philomory we changed how complex flux queries are parsed. it is available in 8.0.0-beta1 or later. it should fix the problem (no data should be "lost" in translation), but the response format might be slightly different. 

if it does not solve your issue completely, please open a new issue (i do not want to mix issues with reports about the "old" behavior with issues with reports about the "new" behavior, to keep issues easy to understand).
